core/src/main/java/hivemall/evaluation/AUCUDAF.java

```diff
@@ -49,35 +51,264 @@
+                StructObjectInspector soi = (StructObjectInspector) parameters[0];
+                this.internalMergeOI = soi;
+                this.aField = soi.getStructFieldRef("a");
+                this.scorePrevField = soi.getStructFieldRef("scorePrev");
+                this.fpField = soi.getStructFieldRef("fp");
+                this.tpField = soi.getStructFieldRef("tp");
+                this.fpPrevField = soi.getStructFieldRef("fpPrev");
+                this.tpPrevField = soi.getStructFieldRef("tpPrev");
+            }
+
+            // initialize output
+            final ObjectInspector outputOI;
+            if (mode == Mode.PARTIAL1 || mode == Mode.PARTIAL2) {// terminatePartial
+                outputOI = internalMergeOI();
+            } else {// terminate
+                outputOI = PrimitiveObjectInspectorFactory.writableDoubleObjectInspector;
+            }
+            return outputOI;
+        }
+
+        private static StructObjectInspector internalMergeOI() {
+            ArrayList<String> fieldNames = new ArrayList<String>();
+            ArrayList<ObjectInspector> fieldOIs = new ArrayList<ObjectInspector>();
+
+            fieldNames.add("a");
+            fieldOIs.add(PrimitiveObjectInspectorFactory.writableDoubleObjectInspector);
+            fieldNames.add("scorePrev");
+            fieldOIs.add(PrimitiveObjectInspectorFactory.writableDoubleObjectInspector);
+            fieldNames.add("fp");
+            fieldOIs.add(PrimitiveObjectInspectorFactory.writableLongObjectInspector);
+            fieldNames.add("tp");
+            fieldOIs.add(PrimitiveObjectInspectorFactory.writableLongObjectInspector);
+            fieldNames.add("fpPrev");
+            fieldOIs.add(PrimitiveObjectInspectorFactory.writableLongObjectInspector);
+            fieldNames.add("tpPrev");
+            fieldOIs.add(PrimitiveObjectInspectorFactory.writableLongObjectInspector);
+
+            return ObjectInspectorFactory.getStandardStructObjectInspector(fieldNames, fieldOIs);
+        }
+
+        @Override
+        public AggregationBuffer getNewAggregationBuffer() throws HiveException {
+            AggregationBuffer myAggr = new ClassificationAUCAggregationBuffer();
+            reset(myAggr);
+            return myAggr;
+        }
+
+        @Override
+        public void reset(AggregationBuffer agg) throws HiveException {
+            ClassificationAUCAggregationBuffer myAggr = (ClassificationAUCAggregationBuffer) agg;
+            myAggr.reset();
+        }
+
+        @Override
+        public void iterate(AggregationBuffer agg, Object[] parameters) throws HiveException {
+            ClassificationAUCAggregationBuffer myAggr = (ClassificationAUCAggregationBuffer) agg;
+
+            if (parameters[0] == null) {
+                return;
+            }
+            if (parameters[1] == null) {
+                return;
+            }
+
+            double score = HiveUtils.getDouble(parameters[0], scoreOI);
```

Comment:
if(score < 0.0d || score > 1.0d) {
   throw new UDFArgumentException("score value MUST be in range [0,1]: " + score);
}